configuration/helm-charts/complex_ouauth_conf.md

```diff
@@ -0,0 +1,279 @@
+
+# Goals
+
+This configuration provide complexe configuration to do SASL_SSL with Oauthbearer configuration between the UI Kafka Client and a Kafka Cluster with jwt and scope validation.
+This use a kafka Strimzi distribution, the only difference is the callback handler class used. You can replace it with the default org.apache or confluent library, or even your own
+
+The prerequesit at thisare:
+ - A kubernet secret with your kafka user/password (given by your authent server for example) :
+
+```yaml
+kind: Secret
+apiVersion: v1
+metadata:
+  name: kafkauiclientsecretadfs
+data:
+  clientid: kafka-ui-kafka-clientId
+  clientsecret: my_user_password
+```
+
+- A user provisionned inside ADFS, that should look like:
+
+```yaml
+  applicationGroup: ADFS--infra-kafka
+  id: kafka-ui-kafka-clientId
+  issuer: https://my_adfs.com/adfs
+  permissions:
+    - audience: api://kafka.jwt
+      scopes:
+        - openid
+        - allatclaims
+    - audience: kafka-ui-kafka-clientId
+      scopes:
+        - openid
+        - allatclaims
+  redirectURI:
+    - https://my-kafka-ui.com/login/oauth2/code/auth0
+```
+
+ - A the user inside kafka that should have a name equal to: ```kafka-ui-kafka-clientId``` and ACL needed to make the Kafka-Ui work [see](https://ui.docs.kafbat.io/quick-start/prerequisites/permissions/required-acls#acls-for-standalone-kafka)
+
+
+- A secret containing the truststore needed for Kafka Broker and Authentication server ssl communications
+
+```yaml
+kind: Secret
+apiVersion: v1
+metadata:
+  name: truststore-secret
+data:
+  truststore.jks: your_trust.jks  
+type: Opaque
+```
+
+# helm Configuration
+
+```yaml
+    spec:
+      containers:
+        - env:
+            - name: KAFKA_CLUSTERS_0_READONLY
+              value: "false"
+            - name: KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS
+              value: broker-kafka-extoauth-bootstrap:9096
+            - name: KAFKA_CLUSTERS_0_ZOOKEEPER
+              value: broker-zookeeper-client:2181
+            - name: KAFKA_CLUSTERS_0_PROPERTIES_ENDPOINT_IDENTIFICATION_ALGORITHM
+              value: ""
+            - name: JAVA_OPTS
+              value: -Djavax.net.ssl.trustStore=/var/run/secrets/truststore.jks -Djavax.net.ssl.trustStorePassword=changeit
+            - name: KAFKA_CLUSTERS_0_SCHEMAREGISTRY
+              value: https://broker-apicurio-schema-registry-service:8443/apis/ccompat/v7
```

Comment:
please note in the guide header that this guide is for apicurio, these are quite specific details of this setup, but still the overall guide might be useful for others, it's just that it would be nice to know what one's dealing with before going deep into the details